apps/web/src/components/modals/failed-ping-alert-confirmation.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "@openstatus/ui";
+
+import { LoadingAnimation } from "@/components/loading-animation";
+import type { MonitorProps } from "../forms/montitor-form";
+
+type FailedPingAlertConfirmationProps = {
+  monitor: MonitorProps;
+  pingFailed: boolean;
+  setPingFailed: React.Dispatch<React.SetStateAction<boolean>>;
+  onConfirm: (props: MonitorProps) => void;
+};
+
+export const FailedPingAlertConfirmation = ({
+  onConfirm: upsertMonitor,
+  pingFailed,
+  setPingFailed,
+  monitor,
+}: FailedPingAlertConfirmationProps) => {
+  const [isPending, startTransition] = React.useTransition();
+  const handleSubmit = () => {
+    startTransition(async () => {
+      upsertMonitor(monitor);
+    });
+    setPingFailed(false);
+  };
```

Comment:
We need to move the `setState` into the transition and add an `await`. Otherwise the modal will close immediately without waiting the save to happen.

You might wanna update the `Props`:

```
onConfirm: (props: MonitorProps) => Promise<void>;
```

```suggestion
  const handleSubmit = () => {
    startTransition(async () => {
      await upsertMonitor(monitor);
      setPingFailed(false);
    });
  };
```